flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -70,6 +71,8 @@
 
     ResourceProfile getFreeResourceOf(InstanceID instanceID);
 
+    Collection<TaskManagerSlotInformation> getAllocatedSlotsOf(InstanceID instanceID);
```

Comment:
This interface exposes `TaskManagerSlotInformation`, which should be internal to slot manager.
I'd suggest to return `SlotInfo` here.